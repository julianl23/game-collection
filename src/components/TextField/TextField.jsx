import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const TextField = ({ id, labelText, type, className }) => {
  const Wrapper = styled.div`
    margin-bottom: 20px;
  `;

  const Label = styled.label`
    display: block;
    margin-bottom: 10px;
  `;

  const Input = styled.input`
    display: block;
    border: 1px solid ${({ theme }) => theme.borderGray};
    background: ${({ theme }) => theme.purpleWhite};
    font-size: ${({ theme }) => theme.fontSizeNormal};
    padding: 15px 10px;
    height: 45px;
    width: 100%;
  `;

  return (
    <Wrapper className={className}>
      <Label htmlFor={id}>{labelText}</Label>
      <Input id={id} type={type} />
    </Wrapper>
  );
};

TextField.propTypes = {
  id: PropTypes.string.isRequired,
  labelText: PropTypes.string.isRequired,
  type: PropTypes.string,
  className: PropTypes.string,
};

TextField.defaultProps = {
  type: "text",
  className: "",
};

export default TextField;
