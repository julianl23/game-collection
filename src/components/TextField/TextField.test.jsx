import React from "react";
import TestRenderer from "react-test-renderer";
import { shallow } from "enzyme";
import TextField from "./TextField";

describe("TextField", () => {
  it("renders correctly", () => {
    const instance = TestRenderer.create(
      <TextField id="hello" labelText="Hello" />
    );
    expect(instance.toJSON()).toMatchSnapshot();
  });

  it("displays the passed in value when one is set", () => {
    const instance = TestRenderer.create(
      <TextField id="hello" labelText="Hello" value="world" />
    );
    expect(instance.toJSON()).toMatchSnapshot();
  });

  it("displays an error message if one is provided", () => {
    const instance = TestRenderer.create(
      <TextField
        id="hello"
        labelText="Hello"
        value="world"
        errorMessage="Field is required"
      />
    );
    expect(instance.toJSON()).toMatchSnapshot();
  });

  it("calls the provided onchange function when it is changed", () => {
    const mock = jest.fn();
    const wrapper = shallow(
      <TextField id="hello" labelText="Hello" onChange={mock} />
    );
    wrapper
      .find("#hello")
      .simulate("change", { target: { value: "My new value" } });
    expect(mock).toHaveBeenCalled();
  });
});
