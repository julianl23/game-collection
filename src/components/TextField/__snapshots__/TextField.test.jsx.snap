// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField renders correctly 1`] = `
.c0 {
  margin-bottom: 20px;
}

.c1 {
  display: block;
  margin-bottom: 10px;
}

.c2 {
  display: block;
  border: 1px solid;
  padding: 15px 10px;
  height: 45px;
  width: 100%;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor="hello"
  >
    Hello
  </label>
  <input
    className="c2"
    id="hello"
    type="text"
  />
</div>
`;
